gobblin-api/src/main/java/org/apache/gobblin/configuration/ConfigurationKeys.java

```diff
@@ -296,6 +296,7 @@ public class ConfigurationKeys {
   public static final String WORK_UNIT_STATE_ACTUAL_HIGH_WATER_MARK_KEY = "workunit.state.actual.high.water.mark";
   public static final String WORK_UNIT_DATE_PARTITION_KEY = "workunit.source.date.partition";
   public static final String WORK_UNIT_DATE_PARTITION_NAME = "workunit.source.date.partitionName";
+  public static final String WORK_UNIT_FAST_FAIL_ENABLED = "workunit.fast.fail.enabled";
```

Comment:
Are there flows that allow failures? i.e. we only want to throw an exception if all workunits fail to generate?

If so, maybe this boolean can be named something like `no_partial_failures` or just `allow_partial_failures` and flip the boolean accordingly.

Reply:
that's correct. We want to avoid the false positives and explicitly fail in case of any execution exceptions happening during the generation of work units. I have renamed the variable to avoid the confusion and updated my PR.